pulsar-broker/src/test/java/org/apache/pulsar/broker/BrokerTestUtil.java

```diff
@@ -44,4 +44,19 @@ public static <T> T spyWithClassAndConstructorArgs(Class<T> classToSpy, Object..
                 .useConstructor(args)
                 .defaultAnswer(Mockito.CALLS_REAL_METHODS));
     }
+
+    /**
+     * Create a Mockito spy that is stub-only which does not record method invocations,
+     * thus saving memory but disallowing verification of invocations.
```

Comment:
> but disallowing verification of invocations

If so, can we replace `spy(obj)` with bare `obj`?

Reply:
no. stubbing is still used in the current solution.

Reply:
Stubbing is used to mock over real methods? 

~~At least for `admin = spy(...)` I don't find a call to mock over real methods.~~ Well. One reference at `org.apache.pulsar.broker.admin.NamespacesTest#testDeleteNamespaceWithBundles` L816.